### PR TITLE
Replacing size hash with multi stage hash dup check

### PR DESCRIPTION
**Pull request by bitsondatadev**
Fix: #9

### CODE REVIEW

File: google_photos_takeout_helper/__main__.py [hunk 1]
```diff
@@ -137,29 +141,29 @@ def find_duplicates(path, filter_fun=lambda file: True):
                     # if the target is a symlink (soft one), this will
                     # dereference it - change the value to the actual target file
                     full_path = _os.path.realpath(full_path)
-                    file_size = _os.path.getsize(full_path)
+                    md5 = hash(full_path)
```
Comment (bitsondatadev) [inline google_photos_takeout_helper/__main__.py:144]: possibly split here
```
file_size = _os.path.getsize(full_path)
hash = file_size
if file_size > 10MB:
    hash = hash(full_path)

duplicate = duplicate_hashes.get(hash)
```

**Comment by bitsondatadev** [comment]
Okay for a small test, I just wanted to verify that this no longer had same size files getting removed and that actual duplicates are still being removed. So I created files 1-5.jpg with an extra 3.jpg in different date folders. and simply put the value 1 in 1.jpg, 2 in 2.jpg, 3 in both 3.jpg files, etc... All files are the same size and with the old setup, it would have chosen one of these files as the original and deleted the rest.

Original Files and contents (notice all are 2B)
```
% ls -lah ~/Desktop/dirs/dir2/*/*.jpg
-rw-r--r--  1 user  staff     2B Jan  1  2016 /Users/user/Desktop/dirs/dir2/2016-01-01/1.jpg
-rw-r--r--  1 user  staff     2B Jan  1  2016 /Users/user/Desktop/dirs/dir2/2016-01-01/2.jpg
-rw-r--r-- 1 user  staff     2B Jan  1  2017 /Users/user/Desktop/dirs/dir2/2017-01-01/3.jpg
-rw-r--r-- 1 user  staff     2B Jan  1  2018 /Users/user/Desktop/dirs/dir2/2018-01-01/3.jpg
-rw-r--r--  1 user  staff     2B Jan  1  2018 /Users/user/Desktop/dirs/dir2/2018-01-01/4.jpg
-rw-r--r-- 1 user  staff     2B Jan  1  2018 /Users/user/Desktop/dirs/dir2/2018-01-01/5.jpg

% cat ~/Desktop/dirs/dir2/*/*.jpg
1
2
3
3
4
5
```
Here is the output folder. One of the 3.jpg files is removed but the other files 1-5.jpg remain in tact.
```
% ls -lah ~/Desktop/dirs/dir3/*.jpg
-rw-r--r--  1 user  staff     2B Jan  1  2016 /Users/user/Desktop/dirs/dir3/1.jpg
-rw-r--r--  1 user  staff     2B Jan  1  2016 /Users/user/Desktop/dirs/dir3/2.jpg
-rw-r--r--  1 user  staff     2B Jan  1  2018 /Users/user/Desktop/dirs/dir3/3.jpg
-rw-r--r--  1 user  staff     2B Jan  1  2018 /Users/user/Desktop/dirs/dir3/4.jpg
-rw-r--r--  1 user  staff     2B Jan  1  2018 /Users/user/Desktop/dirs/dir3/5.jpg
```

**Comment by bitsondatadev** [comment]
@TheLastGimbus just merged in the latest changes. Should be ready to merge.

**Comment by bitsondatadev** [comment]
@TheLastGimbus any update?

**Comment by TheLastGimbus** [comment]
Sorry, I was kinda afraid to merge something that I don't fully understand, but tested it, and it's fully works :heart:  

Script took 20 extra seconds to execute tho, and I'm afraid if people with 10TB of photos will have it running whole day, but this is right thing to do :+1:

**Comment by bitsondatadev** [comment]
I can get on a call with you to explain.

**Comment by TheLastGimbus** [comment]
Noo, no problem, I'm just too lazy to sit and look at this, but it works, is nicely formatted and commented - couldn't wish for a better PR :1st_place_medal:

**Comment by bitsondatadev** [comment]
Haha okay! Well let me know if you need any explanation. I have plenty of ways to get a hold of me on my GitHub profile. 

The performance is an aspect we can attempt to improve later. Most importantly it's correct. When we're getting to the size of TB we're looking at big data level analysis and a day to run a job isn't out of the ordinary 😉.

**Pull request merged**